irr/include/SSkinMeshBuffer.h

```diff
@@ -368,15 +348,11 @@ struct SSkinMeshBuffer : public IMeshBuffer
 	//! Call this after changing the positions of any vertex.
 	void boundingBoxNeedsRecalculated(void) { BoundingBoxNeedsRecalculated = true; }
 
-	std::vector<video::S3DVertexTangents> Vertices_Tangents;
-	std::vector<video::S3DVertex2TCoords> Vertices_2TCoords;
-	std::vector<video::S3DVertex> Vertices_Standard;
-	std::vector<u16> Indices;
+	SVertexBufferTangents *Vertices_Tangents;
+	SVertexBufferLightMap *Vertices_2TCoords;
+	SVertexBuffer *Vertices_Standard;
+	SIndexBuffer *Indices;
```

Comment:
This adds another level of indirection for each `getVertex(u32 index)` call (and others). Was this done on purpose to switch pointers back and forth easily? Otherwise I don't see much of a purpose.

Reply:
I'm aware.
But vertex/index buffers are reference-counted so they need to be on the heap.

Reply:
Unless we'd do inheritance of each - private, not public - but even then, the function and member name collisions would be rather annoying.